common/main/java/com/couchbase/lite/internal/BaseImmutableReplicatorConfiguration.java

```diff
@@ -181,6 +182,22 @@ public void addConnectionOptions(@NonNull Map<String, Object> options) {
         final Map<String, Object> httpHeaders = new HashMap<>();
         // User-Agent:
         httpHeaders.put("User-Agent", CBLVersion.getUserAgent());
+
+        // Cookies:
+        final String cookieName = "Cookies";
+        if ((headers != null) && (!headers.isEmpty()) && headers.containsKey(cookieName)) {
+            //get current cookies value from option to add a new cookie
+            final String current = (String) options.get(C4Replicator.REPLICATOR_OPTION_COOKIES);
+            final StringBuffer cookieStr = (current == null) ? new StringBuffer() : new StringBuffer(current);
```

Comment:
The thing to do here is:
final Object existingCookies = options.get(C4Replicator.REPLICATOR_OPTION_COOKIES);
if (existingCookeis instanceof String) { ...
`null` is never an instance of anything, and doing it this way (an idiom) is type safe